custom-script/utils.py

```diff
@@ -124,25 +146,59 @@ def predict_on_sh(model_script, bbox, size, timestamp, config):
     return request.get_data()[0]
 
 
-def get_predictions(patch, model, config):
-    model_script = parse_model(model, None)
-    sh_prediction = predict_on_sh(model_script, patch.bbox, (64, 64), patch.timestamp[0], config)
-
+def get_model_predictions(patch: EOPatch, model: Any) -> np.ndarray:
     features = patch.data["FEATURES"][0]
-    f_s = features.shape
-    model_prediction = model.predict_proba(features.reshape(f_s[0] * f_s[1], f_s[2]))[..., -1].reshape(f_s[0], f_s[1])
+    height, width, nfeats = features.shape
+    reshaped_features = features.reshape(height * width, nfeats)
+    return model.predict_proba(reshaped_features)[..., -1].reshape(height, width)
 
-    return features[..., [2, 1, 000]], sh_prediction, model_prediction
 
+def get_sh_predictions(patch: EOPatch, model: Any, config: SHConfig) -> np.ndarray:
+    model_script = parse_model(model, None)
+    return predict_on_sh(model_script, patch.bbox, (64, 64), patch.timestamp[0], config)
+
+
+def print_results(
+    f1_scores: np.ndarray,
+    recall: np.ndarray,
+    precision: np.ndarray,
+    predict_labels_test: np.ndarray,
+    labels_test: np.ndarray,
+):
+    class_names = ["non-water", "water"]
+
+    print("Classification accuracy {:.1f}%".format(100 * metrics.accuracy_score(labels_test, predict_labels_test)))
+    print(
+        "Classification F1-score {:.1f}% \n".format(
+            100 * metrics.f1_score(labels_test, predict_labels_test, average="weighted")
+        )
+    )
+    print("             Class              =  F1  | Recall | Precision")
+    print("         --------------------------------------------------")
+    for idx, classname in enumerate(class_names):
+        print(
+            "         * {0:20s} = {1:2.1f} |  {2:2.1f}  | {3:2.1f}".format(
+                classname, f1_scores[idx] * 100, recall[idx] * 100, precision[idx] * 100
+            )
+        )
 
-def plot_comparison(rgb_array, sh_prediction, model_prediction, threshold=0.5):
+
+def plot_comparison(
+    patch: EOPatch,
+    sh_prediction: np.ndarray,
+    model_prediction: np.ndarray,
+    threshold: float = 0.5,
+    factor: float = 3.15,
+):
     fig, ax = plt.subplots(nrows=2, ncols=4, figsize=(24, 12), sharex=True, sharey=True)
 
     for axx in ax.flatten():
         axx.set_xticks([])
         axx.set_yticks([])
 
-    ax[0][0].imshow(3.5 * rgb_array)
+    rgb_image = factor * patch.data["BANDS-S2-L1C"][0][..., [3, 2, 1]].squeeze()
```

Comment:
which axis are you performing the squeeze on?

Reply:
removed